packages/dds/tree/src/test/shared-tree/sharedTree.bench.ts

```diff
@@ -0,0 +1,877 @@
+];
+
+const rootFieldSchema = fieldSchema(FieldKinds.value);
+const globalFieldSchema = fieldSchema(FieldKinds.value);
+const rootNodeSchema = namedTreeSchema({
+	name: brand("TestValue"),
+	localFields: {
+		optionalChild: fieldSchema(FieldKinds.optional, [brand("TestValue")]),
+	},
+	extraLocalFields: fieldSchema(FieldKinds.sequence),
+	globalFields: [globalFieldKey],
+});
+const testSchema: SchemaData = {
+	treeSchema: new Map([[rootNodeSchema.name, rootNodeSchema]]),
+	globalFieldSchema: new Map([
+		[rootFieldKey, rootFieldSchema],
+		[globalFieldKey, globalFieldSchema],
+	]),
+};
+
+// TODO: Once the "BatchTooLarge" error is no longer an issue, extend tests for larger trees.
+describe("SharedTree benchmarks", () => {
+	it("can move nodes across fields", async () => {
+		const [provider, trees] = await createSharedTrees(
+			getTestSchema(FieldKinds.sequence),
+			[{ type: rootSchemaName }],
+			1,
+		);
+		const tree = trees[0];
+		insertNodesToEditableTree(tree, 4, TreeShape.Wide, TestPrimitives.Float);
+	});
+	describe("Direct JS Object", () => {
+		for (let dataType = 0 as TestPrimitives; dataType <= 4; dataType++) {
+			for (const [i, benchmarkType] of nodesCountDeep) {
+				let tree: Jsonable;
+				benchmark({
+					type: benchmarkType,
+					title: `Deep Tree as JS Object (${TestPrimitives[dataType]}): reads with ${i} nodes`,
+					before: async () => {
+						tree = getTestTreeAsJSObject(i, TreeShape.Deep, dataType);
+					},
+					benchmarkFn: () => {
+						readTreeAsJSObject(tree);
+					},
+				});
+			}
+			for (const [i, benchmarkType] of nodesCountWide) {
+				let tree: Jsonable;
+				benchmark({
+					type: benchmarkType,
+					title: `Wide Tree as JS Object (${TestPrimitives[dataType]}): reads with ${i} nodes`,
+					before: async () => {
+						tree = getTestTreeAsJSObject(i, TreeShape.Wide, dataType);
+					},
+					benchmarkFn: () => {
+						readTreeAsJSObject(tree);
+					},
+				});
+			}
+			for (const [i, benchmarkType] of nodesCountDeep) {
+				let tree: Jsonable;
+				benchmark({
+					type: benchmarkType,
+					title: `Deep Tree as JS Object (${TestPrimitives[dataType]}): writes with ${i} nodes`,
+					before: async () => {},
+					benchmarkFn: () => {
+						tree = getTestTreeAsJSObject(i, TreeShape.Deep, dataType);
+					},
+				});
+			}
+			for (const [i, benchmarkType] of nodesCountWide) {
+				let tree: Jsonable;
+				benchmark({
+					type: benchmarkType,
+					title: `Wide Tree as JS Object (${TestPrimitives[dataType]}): writes with ${i} nodes`,
+					before: async () => {},
+					benchmarkFn: () => {
+						tree = getTestTreeAsJSObject(i, TreeShape.Wide, dataType);
+					},
+				});
+			}
+			describe(`Edit JS Object ${TestPrimitives[dataType]}`, () => {
+				for (const [i, benchmarkType] of nodesCountDeep) {
+					let tree: Jsonable;
+					let leafNode: Jsonable;
+					benchmark({
+						type: benchmarkType,
+						title: `Update value at leaf of ${i} deep tree`,
+						before: async () => {
```

Comment:
The before function does not have to be async: the benchmark library explicitly supports both, so you should make it synchronous when possible. THe tests will run slightly faster if you do that, and it makes the call stacks simpler as well as just being more clear about whats happening.

Reply:
removed unused async